### PR TITLE
[QHC-700] Add qibo routing

### PR DESCRIPTION
**Pull request by GuillermoAbadLopez**
## Adding Qibo routing to `qililab`, both for:
 - Star topology: with the Star specific algorithm of qibo
 - And non-Star topologies: With the General ReverseLayout and SabreRouting from https://arxiv.org/abs/1809.02573
 
 ## PROBLEMS TO SOLVE:
 - [x] Automatically undo mapping of qubits in the results? 🤔 
 - [x] Drags are not a qibo native gates, and its Transpiler dunno if handle it well? (✅ -> handles well)
 - [x] Make transpiler loop and keep the best optimization < SWAPS (✅ -> implemented with `_iterate_routing`)
 
## Missing:
 - [x] Better documentation, specifying the algorithms used.
 - [x] Manually testing the function
 - [x] Adding basic unit test for it
 - [x] Simplify unit test, make them more clever with parametrize
 - [ ] Test the results mapping of qubits, and everything else in HW 🔥
 
 -----
 ## More improvements on the transpiler structure, added by: https://github.com/qilimanjaro-tech/qililab/pull/823
 - [x] Adding direct gate cancellations before pulse scheduling
 - [x] Separate drag bunching from Victor?
 - [ ] Missing, removal of 2pi*N, amplitude Drags

### CODE REVIEW

**Comment by fedonman** [comment]
I am starting reviewing now, first thing I noticed is that changelog entry is missing! Please write a detailed one, with an example on how to use the routing.

**Comment by GuillermoAbadLopez** [comment]
> I am starting reviewing now, first thing I noticed is that changelog entry is missing! Please write a detailed one, with an example on how to use the routing.

You are soo right, my bad! Adding it right now 👍

**Comment by GuillermoAbadLopez** [comment]
@fedonman changelog added! 🔥

**Review by fedonman — APPROVED** [review]
Awesome job mate! :100:

File: docs/releases/changelog-dev.md [hunk 1]
```diff
@@ -89,6 +89,57 @@
...
+# Default Transpilation (ReverseTraversal, Sabre and Platform connectivity):
+routed_circ, final_layouts = transpiler.route_circuit([c])
+# With Non-Default Trivial placer, specifying the kwargs, for the router, and different coupling_map:
+routed_circ, final_layouts = transpiler.route_circuit([c], placer=Trivial, router=(Sabre, {"lookahead": 2}, coupling_map=<some_different_topology>))
+# Or finally, Routing with a concrete Routing instance:
+router = Sabre(connectivity=None, lookahead=1) # No connectivity needed, since it will be overwritten by the specified in the Transpiler:
+routed_circ, final_layouts = transpiler.route_circuit([c], placer=Trivial, router=router, coupling_map=<connectivity_to_use>)
+```
+
+[#821](https://github.com/qilimanjaro-tech/qililab/pull/821)
```
Comment (fedonman) [inline docs/releases/changelog-dev.md:141]: If you can indent these, so they belong to the bullet point.

Comment (GuillermoAbadLopez) [reply docs/releases/changelog-dev.md:145]: Done!

**Pull request merged**